sky/shell/platform/android/io/flutter/view/FlutterMain.java

```diff
@@ -176,7 +176,7 @@ public void connectToService(Context context, Core core, MessagePipeHandle pipe)
         registry.register(HapticFeedback.MANAGER.getName(), new ServiceFactory() {
             @Override
             public void connectToService(Context context, Core core, MessagePipeHandle pipe) {
-                HapticFeedback.MANAGER.bind(new HapticFeedbackImpl(), pipe);
+                HapticFeedback.MANAGER.bind(new HapticFeedbackImpl((android.app.Activity) context), pipe);
```

Comment:
The registry is static.  However, the service provider (which references the registry) is per-FlutterView.  It holds the context and passes it along to this callback.